mobile/src/main/java/ch/epfl/sdp/mobile/application/chess/notation/UCINotation.kt

```diff
@@ -0,0 +1,18 @@
+package ch.epfl.sdp.mobile.application.chess.notation
+
+import ch.epfl.sdp.mobile.application.chess.engine.rules.Action
+import ch.epfl.sdp.mobile.application.chess.notation.UCINotationCombinators.uciActionList
+
+/** An object which contains some utilities to transform UCI notation to a [List] of [Action]s */
+object UCINotation {
+
+  /**
+   * Parses a [List] of [Action] from the provided [String] text in UCI notation.
+   *
+   * @param text the [String] that should be parsed.
+   * @return the [List] of [Action] that was found.
+   */
+  fun parseActionList(text: String): List<Action> {
+    return uciActionList().parse(input = text).single().output
+  }
```

Comment:
I suggest changing the return type to `List<Action>?` so the case where the input is not valid is explicitly handled (currently, it will throw an exception on `single()`, which could be changed to `singleOrNull()` or `firstOrNull()` to be possibly more efficient).

Reply:
Addressed in [67a6ba8](https://github.com/epfl-SDP/android/pull/283/commits/67a6ba8e69ab70d1449265ec841c3786d91bddf5).